### ISSUE TITLE
Unmaintained.

### ISSUE THREAD

**Issue by papertigers** [StreamCenter/StreamCenter#91]
Hey wondering why you switched this to "Unmaintained."
Lack of interest?  I am interested in contributing.

**Comment by olivierboucher** [comment]
@papertigers Sorry for the delay, as I mentioned in the README, people are being scums and they copied the app to sell it under a different name. There's no point for me to work on this anymore, since it'll get copied again and sold

**Comment by tjarratt** [comment]
For what it's worth, I'm maintaining a fork right now, although I've made some additional changes that other people may consider undesirable.

https://github.com/tjarratt/StreamCenter/tree/run-locally

Most notably, my television has overscan (and it cannot be disabled for HDMI input), which means that I always lose a fair amount of the edge of any app. I took the liberty of changing my fork of the stream center's UI to accommodate my own television setup.

Just finished updating the client to work with the recent Client-ID header changes that Twitch has finished implementing. If you want to try it out, you'll need to create an app under Twitch's development program (free) and then create an xcconfig file called `StreamCenter/NotMySecretConfigWink.xcconfig` with the correct key in there.

Thanks for creating this @OlivierBoucher -- I'm sorry that you felt backstabbed after going through the effort of building this app, but I'm very glad to have it. This is by far the most used app on my apple tv, with hundreds of hours of uptime.

**Comment by sxnahm** [comment]
Certainly agree with @tjarratt -- I've also been using the streamcenter app quite a bit. The tvOS app's twitch feed appears to have stopped working, so when I went to the customer URL page I see that the streamcenterapp.com site is no longer active. I'm guessing that is why the twitch stream is no longer working.  Does this permanently break this app?

**Comment by tjarratt** [comment]
The twitch API changed such that each HTTP request made to twitch must include a "client id"

In my fork of StreamCenter I've made the changes necessary to support this but it's not as simple as just using my fork. 

One still needs to set their own client-id up with Twitch and provide it when building StreamCenter. One would also need to set up their own code signing identity with Apple, since that should never be shared. 

Would it be helpful for me to provide some tips for how to get the app to compile in the README, or would it be better if I put a version up in the App Store?

Sent From A Very Small Keyboard

**Comment by sxnahm** [comment]
How about both, @tjarratt? :-)

**Comment by tjarratt** [comment]
Because the days are short, and the list of things one **could** do is impossibly long. 

I wasn't aware that Olivier had actually published to the App Store. Will see if I can get in touch with him and see if I can update it, otherwise I'll put up a version myself. 

Will start with updating instructions for building it oneself with Xcode since that's fairly straightforward. 

Sent From A Very Small Keyboard

**Comment by tjarratt** [comment]
Hey @OlivierBoucher I'm interested in deploying a new version of StreamCenter to the AppStore. The one you deployed some time ago no longer works with Twitch, because of [the great ClientID apocalypse](https://discuss.dev.twitch.tv/t/client-id-requirement-faqs/6108).

My current feelings are that it would be easiest for me to just put a new version out. However ...
- It would be slightly difficult to publish an update to the existing app (because you and I would need to coordinate)
- It's rather straightforward for me to put a new version in the App Store
- It's annoying for existing users to need to download a new app
- It's confusing for new users to see two versions in the app store

Do you have any thoughts? I know you were frustrated when someone took all of your effort and sold it for $1, so I wanted to be proactive and ask for your thoughts and concerns first. 

In the interest of transparency, I would probably charge $1 in the app store -- since I've put a nontrivial amount of my own time into this, and I intend to keep working on it.

**Comment by olivierboucher** [comment]
Hey guys - Sorry for the delay. Looking through your posts ATM.

**Comment by olivierboucher** [comment]
@tjarratt Since I've put a bad licence on this in the first place you're free to do what ever you want !

I don't like the idea of this app being sold, but I guess everyone has to pay their bills...

If you want to charge users for this, please create a separate app.

**Comment by olivierboucher** [comment]
@tjarratt However, I may have interest in co-maintaining it. My swift skills are rusty now but I have some time for this.

**Comment by tjarratt** [comment]
@OlivierBoucher awesome! I'll look at submitting a PR in the next few days.

Some of the biggest changes I've made are
- dropping cocoapods for carthage
- adopting test driven development with Quick and Nimble
- switching out singletons for instances that are injected through a constructor

Those are very, very much different from the existing style of the codebase, so I wanted to call that out early, so you aren't too surprised. I'd be happy to explain in more detail why I chose each of those if you'd like, I really do feel that they have helped improve my fork of StreamCenter, albeit in a small way.

**Comment by papertigers** [comment]
@OlivierBoucher @tjarratt I am certainly interested in helping maintain and add features. I started to write just a pure twitch client but haven't found time to finish it. Would be easier working with people on an existing code base. I have a few ideas as well.

**Comment by olivierboucher** [comment]
@papertigers @tjarratt I'm not up to date w/ swift development so I trust you guys. 

Since the api is now shut down, we would need to removed the custom url feature. I don't think it was used that much anyway...

I also have a couple ideas from old emails that were sent to me.

Let's create a slack org to discuss.

**Comment by waterskier2007** [comment]
@papertigers @tjarratt @OlivierBoucher I would be happy to get back in there. I apologize for falling off the face of the earth, but after we had that guy put up a clone, I kind of got discouraged. Don't we have an existing slack organization.

**Comment by papertigers** [comment]
@waterskier2007 @tjarratt @OlivierBoucher I'm fine with slack or discord. Anyone want to set it up or have a preference? If not I can get things rolling.

**Comment by olivierboucher** [comment]
@waterskier2007 Yeah but the org name is something like GithubContributors lol.. Let's get a relevant name and start fresh

**Comment by olivierboucher** [comment]
@waterskier2007 @papertigers @tjarratt got the slack created.

Only invited @tjarratt  because I don't have your emails. You contact me at info@olivierboucher.com to get invited :)

**Comment by papertigers** [comment]
@OlivierBoucher email sent

**Comment by papertigers** [comment]
@tjarratt are you going to send a PR with your changes.  I think what you have is a good starting point.  I prefer Carthage as well.

**Comment by 3van** [comment]
Oh hey, glad to see this is picking up some steam again.

I had already started working on some of this stuff (most notably, the API changes). I'd love to help out.

**Comment by tommynsong** [comment]
I probably can't  contribute any code but I will certainly pay for the app once you guy publish a new version. This is by far my favorite tvOS app for Twitch.
